internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -153,24 +153,24 @@ func NewMSSQLConnector(dsn string) (types.MSSQLConnector, error) {
 // you to specify the newMSSQLConnector explicitly.  Intended to be used in unit
 // tests only.
 func NewSingleUseConnectorWithOptions(
-	logger log.Logger,
-	newMSSQLConnector types.MSSQLConnectorCtor,
-	readPreloginRequest types.ReadPreloginRequestFunc,
-	writePreloginResponse types.WritePreloginResponseFunc,
-	readLoginRequest types.ReadLoginRequestFunc,
-	writeLoginRequest types.WriteLoginResponseFunc,
-	writeError types.WriteErrorFunc,
-	newTdsBuffer types.TdsBufferCtor,
+	setters ...types.ConnectorOption,
```

Comment:
I think if we're naming the type `ConnectorOption` (which is reasonable since it is the "functional options" pattern ) then we should stay consistent and this line should be:
```go
options  ...types.ConnectorOption
```
and then in your loop below you do `option(args)`.  

This is the naming Dave uses in that article.  But unnecessarily introducing a new name for a thing (having  "options" and "setters" which mean the same, in this case) should, as a general principle, always be avoided when possible.

Normally I try avoid naming where we "call" a noun (as in `option(args)`) and opt instead for `verb(noun)`, but in this case it seems reasonable to break the rule because of the pattern name, the type name, and the benefits of consistency.